pkg/api/v1/types.go

```diff
@@ -875,6 +875,8 @@ type PodSpec struct {
 	// NodeSelector is a selector which must be true for the pod to fit on a node
 	NodeSelector map[string]string `json:"nodeSelector,omitempty" description:"selector which must match a node's labels for the pod to be scheduled on that node; see http://releases.k8s.io/HEAD/examples/node-selection/README.md"`
 
+	// ServiceAccount is a depreciated alias for ServiceAccountName.
+	ServiceAccount string `json:"serviceAccount,omitempty" description:"deprecated. Alias for ServiceAccountName."`
```

Comment:
Wouldn't naming this DeprecatedServiceAccount defeat the point of this PR? Or is it only the JSON representation that matters? Do the objects never float around unless serialized by JSON? (Naive questions...)

@bgrant0607 what's your take on the name?